Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs

```diff
@@ -20,8 +23,11 @@ namespace Microsoft.MixedReality.Toolkit.SDK.UX.Pointers
     /// Base Pointer class for pointers that exist in the scene as GameObjects.
     /// </summary>
     [DisallowMultipleComponent]
-    public abstract class BaseControllerPointer : ControllerPoseSynchronizer, IMixedRealityPointer, IMixedRealitySpatialInputHandler
+    public abstract class BaseControllerPointer : ControllerPoseSynchronizer, IMixedRealityPointer, IMixedRealitySpatialInputHandler, IMixedRealityTeleportHandler
     {
+        private static IMixedRealityTeleportSystem teleportSystem = null;
```

Comment:
A lot of these additions doesn't seem like things that would be in a "`BaseControllerPointer`" and are teleport-specific. Why are they here and not in specific pointer implementations?

Reply:
They are here because pointers still need to know about teleportation. That's how we're hiding/showing the pointers and their cursors when a teleport request is raised.

Reply:
Right, but hiding/showing pointers depending on the state of other pointers seems like a common scenario and could get out of hand if every pointer needs to be written into the `BaseControllerPointer`. Teleportation isn't necessarily a special case, though I suppose we can take another look in beta.

Reply:
Indeed, and maybe that can be abstracted out a bit more.